pkg/defaults/defaults_global.go

```diff
@@ -0,0 +1,88 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package defaults
+
+import (
+	"os"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/defaults"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func GetglobalBool(cmd *cobra.Command, key string) (bool, error) {
+	globalBoolMap := map[string]bool{
+		"debug":             false,
+		"debug-full":        false,
+		"insecure-registry": false,
+	}
+	if cmd.Flags().Changed(key) {
+		return cmd.Flags().GetBool(key)
+	}
+	vp := viper.GetViper()
+	configBool := vp.Get(key)
+	if configBool != nil {
+		return configBool.(bool), nil
+	} else {
+		if val, ok := globalBoolMap[key]; ok {
+			return val, nil
+		} else {
+			return false, nil
+		}
+	}
+}
+
+func GetglobalString(cmd *cobra.Command, key string) (string, error) {
```

Comment:
Should the following calls use `GetglobalString()`?

https://github.com/ningmingxiao/nerdctl/blob/18f1151de3fa320eda8dde4cf3403d3bde100e26/cmd/nerdctl/logs.go#L62-L65

https://github.com/ningmingxiao/nerdctl/blob/18f1151de3fa320eda8dde4cf3403d3bde100e26/cmd/nerdctl/top.go#L88-L91

Reply:
done